client/src/components/modals/ComponentModal.tsx

```diff
@@ -0,0 +1,278 @@
+								variant="filled"
+								severity="error"
+								action={
+									<IconButton
+										aria-label="close"
+										color="inherit"
+										size="small"
+										onClick={() => {
+											setErrorBar({ ...errorBar, show: false });
+										}}
+									>
+										<CloseIcon fontSize="inherit" />
+									</IconButton>
+								}
+							>
+								{errorBar.message}
+							</Alert>
+						)}
+						<Stack direction="row" spacing={3}>
+							<TextField
+								InputLabelProps={{ shrink: editMode }}
+								type="String"
+								value={name}
+								onChange={(e) => handleChange(e, setName)}
+								fullWidth
+								size="small"
+								id="component-name"
+								label="Name"
+								variant="outlined"
+							/>
+						</Stack>
+						<Stack direction="row" spacing={3}>
+							<TextField
+								InputLabelProps={{ shrink: editMode }}
+								type="String"
+								value={details}
+								onChange={(e) => handleChange(e, setDetails)}
+								fullWidth
+								size="small"
+								id="component-name"
+								label="Details"
+								variant="outlined"
+							/>
+						</Stack>
+						<Stack direction="row" spacing={2}>
+							<FormControl fullWidth variant="filled">
+								<InputLabel id="component-source-label">Telemetry Source</InputLabel>
+								<Select
+									id="component-source"
+									variant="filled"
+									multiple
+									fullWidth
+									value={sourceTypes}
+									onChange={(e) => handleChange(e, setSourceTypes)}
+									input={<OutlinedInput id="select-multiple-chip" label="Chip" />}
+									renderValue={(selected: string[]) => (
+										<Box
+											sx={{
+												display: 'flex',
+												flexWrap: 'wrap',
+												gap: 0.5,
+												transform: 'translateY(20%)'
+											}}
+										>
+											{selected.map((value: string) => (
+												<Chip key={value} label={value} />
+											))}
+										</Box>
+									)}
+									sx={{
+										'& .MuiOutlinedInput-root.Mui-focused .MuiOutlinedInput-notchedOutline': {
+											border: 'none !important'
+										}
+									}}
+									MenuProps={MenuProps}
+									labelId="component-source-label"
+									label=""
+								>
+									<MenuItem key="LORA" value="LORA">
+										LORA
+									</MenuItem>
+									<MenuItem key="APRS" value="APRS">
+										APRS
+									</MenuItem>
+								</Select>
+							</FormControl>
+						</Stack>
+						<Stack direction="row" spacing={1}>
+							<input
+								accept="json/*"
+								hidden
+								className={'button'}
+								id="contained-button-file"
+								onChange={onUploadFile}
+								type="file"
+							/>
+							<label style={{ width: '95%' }} htmlFor="contained-button-file">
+								<Button
+									variant="contained"
+									component="span"
+									className={'button'}
+									size="large"
+									startIcon={<CloudUpload />}
+									fullWidth
+								>
+									Data Configuration
```

Comment:
```suggestion
									{!configFile && "Data Configuration"}
```
So it won't squish the new textbox